caravel/viz.py

```diff
@@ -133,14 +133,20 @@ def get_df(self, query_obj=None):
         self.results = None
 
         # The datasource here can be different backend but the interface is common
+        # timestamp_format is not necessary for Druid side, so when the datasource
+        # is not table, timestamp_format is set as None
+        timestamp_format = None
+        if self.datasource.type == 'table':
+            timestamp_format = self.datasource.timestamp_format
         self.results = self.datasource.query(**query_obj)
         self.query = self.results.query
         df = self.results.df
         if df is None or df.empty:
             raise Exception("No data, review your incantations!")
         else:
             if 'timestamp' in df.columns:
-                df.timestamp = pd.to_datetime(df.timestamp, utc=False)
+                df.timestamp = pd.to_datetime(
+                    df.timestamp, utc=False, format=timestamp_format)
```

Comment:
I like this here as opposed to relying on pandas's `to_datetime` magic, but it breaks down if we need to use a database side `CAST` or `TO_DATE` of some form